syntax = "proto3";
package data_relay_grpc.blob_relay.smoke_test.proto;

// empty message
message Void {}

// unknown error was occurred.
message UnknownError {
    // the error message.
    string message = 1;
}

// Create blob relay session
message CreateSessionRequest {
    // no special member
}
message CreateSessionResponse {
    reserved 1 to 10;

    // the response body.
    oneof result {
        // request is successfully completed.
        uint64 session_id = 12;

        // unknown error was occurred.
        UnknownError unknown_error = 11;
    }
}

// Dispose blob relay session
message DisposeSessionRequest {
    // session id of the target session.
    uint64 session_id = 1;
}
message DisposeSessionResponse {
    reserved 1 to 10;

    // the response body.
    oneof result {
        // request is successfully completed.
        Void success = 12;

        // unknown error was occurred.
        UnknownError unknown_error = 11;
    }
}

// Support functionality for BlobRelaySmokeTest
service BlobRelaySmokeTestSupport {

    // Create blob relay session
    rpc CreateSession(CreateSessionRequest) returns (CreateSessionResponse);

    // Upload BLOB data.
    rpc DisposeSession(DisposeSessionRequest) returns (DisposeSessionResponse);
}
